src/shared/seccomp-util.c

```diff
@@ -869,20 +849,21 @@ int seccomp_restrict_namespaces(unsigned long retain) {
                                 break;
                         }
 
-                        if (clone_reversed_order == 0)
+                        if (CLONE_REVERSED_ORDER)
                                 r = seccomp_rule_add_exact(
                                                 seccomp,
                                                 SCMP_ACT_ERRNO(EPERM),
                                                 SCMP_SYS(clone),
                                                 1,
-                                                SCMP_A0(SCMP_CMP_MASKED_EQ, f, f));
+                                                SCMP_A1(SCMP_CMP_MASKED_EQ, f, f));
                         else
                                 r = seccomp_rule_add_exact(
                                                 seccomp,
                                                 SCMP_ACT_ERRNO(EPERM),
                                                 SCMP_SYS(clone),
                                                 1,
-                                                SCMP_A1(SCMP_CMP_MASKED_EQ, f, f));
+                                                SCMP_A0(SCMP_CMP_MASKED_EQ, f, f));
+
```

Comment:
hmmm, this is a strange check, no? you check via the runtime construct `if` for a compile-time macro `CLONE_REVERSED_ORDER`. Wouldn't it be more obvious to check with the compile-time construct `#if` for the compile-time macro?

i.e. use compile time constructs for compile time macros, but runtime constructs for runtime variables?